packages/wallet/src/TxBuilder/OutputBuilder.ts

```diff
@@ -83,20 +53,12 @@ export class ObservableWalletTxOutputBuilder implements OutputBuilder {
     return this;
   }
 
-  async build(): Promise<MaybeValidTxOut> {
+  build(): MaybeValidTxOut {
```

Comment:
`MaybeValidTxOut` here isn't  great since we're not doing full validation. How about rolling back changes to `OutputBuilder` and doing sth like:

```
interface TxBuilder {
  addOutput(TxOut | OutputBuilder);
}
```

Then we can call the async `build` internally when `TxBuilder.build` is called.

Reply:
So users could either run async `OutputBuilder.build()` themselves if they want to check the validity, or pass the output directly into TxBuilder, which will call `OutputBuilder.build()` when building the transaction?

Reply:
> So users could either run async OutputBuilder.build() themselves if they want to check the validity, or pass the output directly into TxBuilder, which will call OutputBuilder.build() when building the transaction?

Yes, that's the idea

> If so, we could rename OutputBuilder.build() to OutputBuilder.validate()

Not sure about this - it can still be used to build an output (use case isn't limited to just validating)

Reply:
Done in [9165582](https://github.com/input-output-hk/cardano-js-sdk/pull/427/commits/9165582d01e8eccd409d731d88a40e39905f4570)